src/main/java/org/apache/pulsar/manager/service/impl/BrokerStatsServiceImpl.java

```diff
@@ -131,7 +132,26 @@ private void scheduleCollectStats() {
             clusterLists.forEach((clusterMap) -> {
                 String cluster = (String) clusterMap.get("cluster");
                 Pair<String, String> envCluster = Pair.of(env.getName(), cluster);
-                collectStatsServiceUrls.put(envCluster, (String) clusterMap.get("serviceUrl"));
+                String webServiceUrl = (String) clusterMap.get("serviceUrl");
+                if (webServiceUrl.contains(",")) {
+                    String[] webServiceUrlList = webServiceUrl.split(",");
+                    if (StringUtils.isNotBlank(pulsarJwtToken)) {
+                        header.put("Authorization", String.format("Bearer %s", pulsarJwtToken));
+                    }
+                    for (String url : webServiceUrlList) {
+                        if (!url.contains("http://")) {
+                            url = "http://" + url;
+                        }
+                        String httpTestResult = HttpUtil.doGet( url + "/metrics", header);
```

Comment:
Sorry for late reply.
It looks like we are going to call metrics endpoint twice.
Isn't there any way to download such data only once?

Like pushing down the list of endpoints to downstream processing?

Reply:
@eolivelli I changed the interface to call health check, but it still needs several HTTP calls. I haven't found a way to get all the broker status at once. Now, this call is in a scheduled task, and it will run every 5 minutes.